contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity 0.4.24;
+import "../../libraries/SafeMath.sol";
+import "../../libraries/Message.sol";
+import "../BasicBridge.sol";
+import "../../upgradeability/EternalStorage.sol";
+import "../../IBlockReward.sol";
+import "../../ERC677Receiver.sol";
+import "../BasicHomeBridge.sol";
+import "../ERC677Bridge.sol";
+
+
+contract HomeBridgeErcToNative is EternalStorage, BasicBridge, BasicHomeBridge {
+
+    function () public payable {
+        require(msg.value > 0);
+        require(msg.data.length == 0);
+        require(withinLimit(msg.value));
+        setTotalSpentPerDay(getCurrentDay(), totalSpentPerDay(getCurrentDay()).add(msg.value));
+        setTotalBurntCoins(totalBurntCoins().add(msg.value));
+        address(0).transfer(msg.value);
+        emit UserRequestForSignature(msg.sender, msg.value);
+    }
+
+    function initialize (
+        address _validatorContract,
+        uint256 _dailyLimit,
+        uint256 _maxPerTx,
+        uint256 _minPerTx,
+        uint256 _homeGasPrice,
+        uint256 _requiredBlockConfirmations,
+        address _blockReward
+
+    ) public returns(bool)
+    {
+        require(!isInitialized());
+        require(_validatorContract != address(0));
+        require(_homeGasPrice > 0);
+        require(_requiredBlockConfirmations > 0);
+        require(_minPerTx > 0 && _maxPerTx > _minPerTx && _dailyLimit > _maxPerTx);
+        require(_blockReward != address(0) && isContract(_blockReward));
```

Comment:
As per specific of the network it is assumed that the address of the block reward contract could be unknown at the phase of bridge contracts initialization. That's why consider to allow set up `blockRewardContract` in both cases:
1. `_blockReward` is zero;
2. `_blockReward` is not zero and it points to a real smart contract. 

It will cause the follow modification in other parts:
1. `blockRewardContract().addExtraReceiver(_value, _recipient);` must be invoked only in the case if `blockRewardContract` is not `address(0)`.
2. a special script must be created to set the block reward contract address by `setBlockRewardContract`. This script will be invoked manually as soon as the address will be generated during the network deployment procedure.